src/02-room-details.js

```diff
@@ -25,7 +25,24 @@ const exampleRoomData = require("../data/rooms");
  *  getRoomByDinosaurName(dinosaurs, rooms, "Pterodactyl");
  *  //> "Dinosaur with name 'Pterodactyl' cannot be found."
  */
-function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
+function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {
+  let results = ""
+  let roomHelp = []
+ for (let i = 0; i < dinosaurs.length;i++){
+   for (let j = 0; j < rooms.length; j++){
+   if (dinosaurName !== dinosaurs[i].name){
+     results = `Dinosaur with name '${dinosaurName}' cannot be found.`
+   }
+     if (dinosaurs[i].dinosaurId !== rooms[j].dinosaurs.includes(dinosaurName)) {
+  results =  `Dinosaur with name '${dinosaurName}' cannot be found.`
+}
+    if (dinosaurs[i].name === dinosaurName){
+       roomHelp.push(dinosaurs[i].dinosaurId)
+       results = roomHelp
+  }
+}return results
+ }return results //"name of the room where dinosaur can be found"
+ }
```

Comment:
It looks like there is some confusion with control flow. Since the dinosaur name is passed to the function and you need the ID to check what rooms it is on, you first want to look through the entire dino array to find the dinosaur. If it is not found **after looping the entire array** then you want to return the appropriate error message.

After you have the ID, then you want to loop the `rooms` array to find what rooms it is in. 

So in this case, you dont need a nested loop but two different loops.